### DISCUSSION TITLE
Set Plot title using nexus format

### DISCUSSION THREAD

**Discussion by Flydroid** [silx-kit/h5web#1700] [Q&A] [answered]
Hi,

I have been testing the nexus format and h5web to display measurement data with correct time axis and labels. 
![image](https://github.com/user-attachments/assets/048d8cad-c132-4d99-a5bf-f4bf4595ba4e)

The x-axis is defined with @axes and the y-data with @signal . This results in that h5web uses the @long_name of "y" as both the y_label and the plot title. 
Is there a way to specify the plot title (or the y-axsis label) seperatly?

Here is the h5 file zipped for upload:
[simple_with_hardlink.zip](https://github.com/user-attachments/files/16710578/simple_with_hardlink.zip)

**Comment by loichuder** [comment, marked as answer]
Hi @Flydroid 

You can add a `title` dataset to your NXData group containing the string you want as title. 

You can find an example of this in the [Mock data demo](https://h5web.panosc.eu/mock) (path `/nexus_entry/nx_process/nx_data`)

Hope this helps !

**Comment by Flydroid** [comment]
Thanks for the answer @loichuder. 
I was looking for an attribute related to title. 
Just out of curiosity: Is there a reason why a dataset is used instead of a "title" attribute to NXdata?

**Reply by loichuder** [reply]
> Just out of curiosity: Is there a reason why a dataset is used instead of a "title" attribute to NXdata?

No clue 😂.

We are following the NeXus specification and it seems to hold a lot of history so that little inconsistencies like this are common.

Btw, the NeXus specification for NXdata is here in case you need it for future reference: https://manual.nexusformat.org/classes/base_classes/NXdata.html

**Reply by axelboc** [reply]
>  Just out of curiosity: Is there a reason why a dataset is used instead of a "title" attribute to NXdata?

It would certainly make our code simpler :joy:
